### DISCUSSION TITLE
Modelling guidelines?

### DISCUSSION THREAD

**Discussion by pbusi** [Open-Orange-Button/Orange-Button-Taxonomy#329] [General]
@mlunelli @NiSHoW

Hi. My company is partner in a EU-funded project concerning photovoltaic systems' operation and maintenance. Our current task is to define APIs for interoperability among various platforms. Orange Button seems to be a candidate for the data modelling part of this task, but, after a preliminary analysis, we have questions about how to a plant should be modeled. I see, from a previous discussion, that apparently there is no publicly available example, which would have helped a lot. I was wondering if anybody could make a toy example and show how it would be represented in OB (say, a plant with one array with one string, one inverter for the latter, one weather sensor for the entire plant, ...). We are also very happy to have a chat with somebody who has the time and is willing to answer our questions. Thank you very much!

**Comment by cwhanse** [comment]
@pbusi @mlunelli @NiSHoW Great question.  Are you wanting to describe the static data (e.g. product, serial number, coordinates), time-series data (e.g., voltage, power, irradiance), or both?

**Reply by pbusi** [reply]
Thanks @cwhanse for your help! As a matter of fact, we are looking for both. We are approaching integration by means of on-demand queries, so typically on static data (essentially, what composes the plant and information concerning specific assets), and stream of events on pub/sub channels - time-series as well as monitoring reports, alarms, and so on. Any example or suggestion would be greatly appreciated.

**Comment by jrippingale** [comment]
Here is an example of a residential Site with construction information that
isn't a perfect match, and it will give you a sense for some of your
questions.  You could also join the working group (Tuesdays at 11a PT) to
learn more so that you can see how to generate JSON for any element of
interest.  I would need your email to add you to the meeting.

**Reply by pbusi** [reply]
Thanks a lot @jrippingale - but I cannot see any attachment. 

However, it may be me not familiar with GitHub - please advise. On the same matter (i.e. my ignorance...), how do I send you my email address privately? I'm going to see if there's a way to chat with you directly.

**Comment by jrippingale** [comment]
Feel free to reach out to me at ***@***.*** directly.  Then I
will resend the attachment! :-)
